docs/source/column-metadata.rst

```diff
@@ -0,0 +1,185 @@
+***************
+Column Metadata
+***************
+
+This document is heavily inspired by the KDB+ DBMS and its driver language ``q``.
+
+In ``q``, tables are constructed like this:
+
+   .. code-block:: sh
+
+      q) t: ([] name: `Alice`Bob`Joe; amount: 100 200 -400)
+
+Primary Keys
+============
+To use a primary key, we construct the table by filling in the square brackets.
+
+   .. code-block:: sh
+
+      q) t: ([id: 1 2 3] name: `Alice`Bob`Joe; amount: 100 200 -400)
+
+Foreign Keys
+============
+Many (if not all) database systems provide a way to relate columns in possibly different tables to each other. When a column in one table refers to a column in another table we call this a *foreign key*.
+
+Foreign keys are indicated by using a table's name (in this case ``t``) as the type of a field when constructing another table:
+
+   .. code-block:: sh
+
+      q) s: ([]; t_id: `t$1 2 3 1 2; classes: `a`b`c`d`e; number: 1.0 2.0 0n 0n 5.0)
+
+
+This allows one to do things like an implicit join, referencing any column in the table referred to by the foreign key:
+
+   .. code-block:: sh
+
+      q) select t_id.name from s
+      name
+      ----
+      b
+      c
+      a
+      b
+      c
+
+
+Other Metadata
+==============
+
+Additionally, KDB+ exposes metadata regarding the structure of a particular column. In particular it allows one to attribute different properties to a column. Those properties are:
```

Comment:
Might also want to look at SQL constraints http://www.w3schools.com/sql/sql_constraints.asp